p2p/net/swarm/addr/addr.go

```diff
@@ -188,10 +190,25 @@ func ResolveUnspecifiedAddresses(unspecAddrs, ifaceAddrs []ma.Multiaddr) ([]ma.M
 	return outputAddrs, nil
 }
 
+const interfaceAddrCacheLife = time.Second * 5
+
+var interfaceAddrLastCache time.Time
+var interfaceAddrCache []ma.Multiaddr
+var interfaceAddrLock sync.Mutex
```

Comment:
:-1: to static caches like this. make it an object instead:

```go
netAddrCache := NewInterfaceAddrCache()
netAddrCache.InterfaceAddresses()
```
and have a "DefaultCache" -- the way `http` lib does. compartmentalizes and cleans all this stuff.